pilot/pkg/model/service.go

```diff
@@ -1118,6 +1118,12 @@ func (s *Service) GetAddresses(node *Proxy) []string {
 // GetAddressForProxy returns a Service's address specific to the cluster where the node resides
 func (s *Service) GetAddressForProxy(node *Proxy) string {
 	if node.Metadata != nil {
+		// pilot/pkg/serviceregistry/kube/conversion.go was updated to remove
+		// DefaultAddress from ClusterVIPs. We need to handle that here in
+		// order to build virtualHost Domains correctly for dual stack
+		if s.DefaultAddress != "" && s.DefaultAddress != constants.UnspecifiedIP {
+			return s.DefaultAddress
+		}
```

Comment:
Not sure I am following this?

Reply:
Will likely rework this anyway in light of your other comments.